### PR TITLE
Fixed typos in `InferenceClient` hyperlinks

### PR DESCRIPTION
**Pull request by sergiopaniego**
I reviewed the newly added [Image-Text to Text](https://huggingface.co/docs/api-inference/en/tasks/image-text-to-text) task and identified some typos in the hyperlinks. This PR fixes those typos.

@merveenoyan

### CODE REVIEW

File: docs/api-inference/tasks/automatic-speech-recognition.md [hunk 1]
```diff
@@ -64,7 +64,7 @@ def query(filename):
 output = query("sample1.flac")
 ```
 
-To use the Python client, see `huggingface_hub`'s [package reference](https://huggingface.co/docs/huggingface_hub/package_reference/inference_client#huggingface_hub.InferenceClient.automatic_speech-recognition).
+To use the Python client, see `huggingface_hub`'s [package reference](https://huggingface.co/docs/huggingface_hub/package_reference/inference_client#huggingface_hub.InferenceClient.automatic_speech_recognition).
```
Comment (pcuenca) [inline docs/api-inference/tasks/automatic-speech-recognition.md:67]: Ah, anchors were wrong!

File: docs/api-inference/tasks/image-text-to-text.md [hunk 2]
```diff
@@ -86,7 +86,7 @@ for chunk in stream:
     print(chunk.choices[0].delta.content, end="")
 ```
 
-To use the Python client, see `huggingface_hub`'s [package reference](https://huggingface.co/docs/huggingface_hub/package_reference/inference_client#huggingface_hub.InferenceClient.image_text-to-text).
+To use the Python client, see `huggingface_hub`'s [package reference](https://huggingface.co/docs/huggingface_hub/package_reference/inference_client#huggingface_hub.InferenceClient.image_text_to_text).
```
Comment (pcuenca) [inline docs/api-inference/tasks/image-text-to-text.md:89]: For some reason I'm not seeing this one yet.

**Review by pcuenca — APPROVED** [review]

**Comment by pcuenca** [comment]
Ah missed that, thank you @hanouticelina!

File: docs/api-inference/tasks/image-text-to-text.md (hunk 2 above)
Comment (sergiopaniego) [reply docs/api-inference/tasks/image-text-to-text.md]: I think that the functionality is still not developed but the pointer is there 😄

**Comment by sergiopaniego** [comment]
> Hi @sergiopaniego thanks a lot for your PR! the `docs/api-inference/tasks/*.md` files are auto-generated by running the `generate.ts` script in `scripts/api-inference`, so we should update the script instead of modifying the .md files directly.
> 
> the issue comes from these lines: [scripts/api-inference/scripts/generate.ts#L443-L444](https://github.com/huggingface/hub-docs/blob/main/scripts/api-inference/scripts/generate.ts#L443-L444) [scripts/api-inference/scripts/generate.ts#L514-L515](https://github.com/huggingface/hub-docs/blob/main/scripts/api-inference/scripts/generate.ts#L514-L515) the current `replace()` method only replaces the first occurrence of the pattern (that's why "image-text-to-text" becomes "image_text-to-text" instead of "image_text_to_text"). to fix this, we can use replaceAll() instead:
> 
> ```diff
> - taskSnakeCase: baseName.replace("-", "_"),
> - taskAttached: baseName.replace("-", ""),
> + taskSnakeCase: baseName.replaceAll("-", "_"),
> + taskAttached: baseName.replaceAll("-", ""),
> ```
> 
> this change will also ensure all huggingface.js package reference links are properly formatted! could you update the script with these changes instead? No need to run the generate script yourself as we have an automated PR that regenerates the documentation every day. Let me know if you need any help with this! 🤗

Wow 😮, thanks a lot for the in-depth explanation @hanouticelina!!! 
I've made the updates following your advice. I hope they meet the requirements 🤗 
Thanks, @pcuenca too!!

**Review by hanouticelina — APPROVED** [review]
all good! thanks a lot @sergiopaniego for this PR 🤗

**Pull request merged**